lib/pdk/util/changelog_generator.rb

```diff
@@ -0,0 +1,113 @@
+        PDK.logger.info _('Determing the target version from CHANGELOG.md')
+
+        # Grab all lines that start with ## between from the latest changes
+        # For example given the changelog below
+
+        # ```
+        # # Change log
+        #
+        # All notable changes to this project will be documented in this file.
+        #
+        # ## [v4.0.0](https://github.com/puppetlabs/puppetlabs-inifile/tree/v4.
+        #
+        # [Full Changelog](https://github.com/puppetlabs/puppetlabs-inifile/com   --+
+        #                                                                           |
+        # ### Changed                                                               |
+        #                                                                           |
+        # - pdksync - FM-8499 - remove ubuntu14 support [\#363](https://github.     |  It's this piece of text we are interested in
+        #                                                                           |
+        # ### Added                                                                 |
+        #                                                                           |
+        # - FM-8402 add debian 10 support [\#352](https://github.com/puppetlabs     |
+        #                                                                           |
+        # ## [v3.1.0](https://github.com/puppetlabs/puppetlabs-inifile/tree/v3.     |
+        #                                                                         --+
+        # [Full Changelog](https://github.com/puppetlabs/puppetlabs-inifile/com
+        #
+        # ### Added
+        #
+        # - FM-8222 - Port Module inifile to Litmus [\#344](https://github.com/
+        # - \(FM-8154\) Add Windows Server 2019 support [\#340](https://github.
+        # - \(FM-8041\) Add RedHat 8 support [\#339](https://github.com/puppetl
+        # ````
+        data = ''
+        in_changelog_entry = false
+        changelog_content.each_line do |line|
+          line.strip!
+          if line.start_with?('[')
+            # We're leaving the latest changes so we can break
+            break if in_changelog_entry
+            in_changelog_entry = true
+          end
+          if in_changelog_entry && line.start_with?('##')
+            data += line
+          end
+        end
+
+        # Check for meta headers in first two header line matches
+        if data =~ %r{^### Changed}
+          # Major Version bump
+          version[0] += 1
+          version[1] = 0
+          version[2] = 0
+        elsif data =~ %r{^### Added}
+          # Minor Version bump
+          version[1] += 1
+          version[2] = 0
+        elsif data =~ %r{^### Fixed}
+          # Patch Version bump
+          version[2] += 1
+        end
+
+        version.join('.')
+      end
+
+      def self.changelog_file
+        @changelog_file ||= PDK::Util::Filesystem.expand_path('CHANGELOG.MD')
```

Comment:
Probably need to look for `CHANGELOG.md` as well for case sensitive filesystems as it's pretty common to use the lowercase extension (or maybe use `fnmatch` with casefold)

Reply:
Looks like the default is `CHANGELOG.md` so 👍 

https://github.com/github-changelog-generator/github-changelog-generator/blob/73f7a9a187c1f2d6a62b290cfcb984aacbbc45f8/lib/github_changelog_generator/parser.rb#L52

Reply:
We're at the mercy of the changelog generator rake task here :-(

Reply:
Fixed